### DISCUSSION TITLE
How do I update main branch data?

### DISCUSSION THREAD

**Discussion by jpnws** [planetscale/discussion#333] [Q&A] [answered]
I currently have main branch in production mode.

If I have data that's on a non-main branch, and roll it into main, is it supposed to copy all the data from it onto the main branch?

**Comment by jpnws** [comment]
Also, how do I reset (delete all) the data that's already on the main/production branch?

**Comment by fooforge** [comment, marked as answer]
Hey @jpnws,

> If I have data that's on a non-main branch, and roll it into main, is it supposed to copy all the data from it onto the main branch?

[Deploy requests](https://planetscale.com/docs/concepts/deploy-requests) enable you to create or revert non-blocking schema changes, but not to keep data in sync between branches. Think of your [branches](https://planetscale.com/docs/concepts/branching) as completely isolated databases that you need to populate with data individually, and of any other branches than your production branch to be ephemeral in nature.

That being said, we have a [feature named Data Branching™](https://planetscale.com/docs/concepts/data-branching) which enables you to seed data from a recent backup when standing up a new branch. It's available from the [Team plan](https://planetscale.com/pricing) and up.

> Also, how do I reset (delete all) the data that's already on the main/production branch?

The easiest would be to change the default branch on your database's settings page allowing you to delete the branch in question, and to stand up a new branch. A more programmatic way would be to delete any rows in your tables and to then drop all tables through a deploy request.
